test/powershell/Language/Operators/TernaryOperator.Tests.ps1

```diff
@@ -0,0 +1,185 @@
+            $ers = $null
+            $result = [Parser]::ParseInput('$true ? :', [ref]$null, [ref]$ers)
+            $ers.Count | Should -Be 2
+
+            $ers[0].IncompleteInput | Should -BeFalse
+            $ers[0].ErrorId | Should -BeExactly 'ExpectedValueExpression'
+            $ers[1].IncompleteInput | Should -BeTrue
+            $ers[1].ErrorId | Should -BeExactly 'ExpectedValueExpression'
+
+            $expr = $result.EndBlock.Statements[0].PipelineElements[0].Expression
+            $expr | Should -BeOfType 'System.Management.Automation.Language.TernaryExpressionAst'
+            $expr.IfTrue | Should -BeOfType 'System.Management.Automation.Language.ErrorExpressionAst'
+            $expr.IfFalse | Should -BeOfType 'System.Management.Automation.Language.ErrorExpressionAst'
+
+            $ers = $null
+            $result = [Parser]::ParseInput('$true ? : 3', [ref]$null, [ref]$ers)
+            $ers.Count | Should -Be 1
+
+            $ers.IncompleteInput | Should -BeFalse
+            $ers.ErrorId | Should -BeExactly "ExpectedValueExpression"
+            $expr = $result.EndBlock.Statements[0].PipelineElements[0].Expression
+            $expr | Should -BeOfType 'System.Management.Automation.Language.TernaryExpressionAst'
+            $expr.IfTrue | Should -BeOfType 'System.Management.Automation.Language.ErrorExpressionAst'
+            $expr.IfFalse | Should -BeOfType 'System.Management.Automation.Language.ConstantExpressionAst'
+        }
+    }
+
+    Context "Using of ternary operator" {
+        BeforeAll {
+            $testCases_1 = @(
+                ## Condition: variable and constant expressions
+                @{ Script = { $true ? 1 : 2 };  ExpectedValue = 1 }
+                @{ Script = { $true? ?1 :2 };   ExpectedValue = 2 }
+                @{ Script = { ${true}?1:2 };    ExpectedValue = 1 }
+                @{ Script = { 1 ? 1kb : 0xf };  ExpectedValue = 1kb }
+                @{ Script = { 0 ?1kb:0xf };     ExpectedValue = 15 }
+                @{ Script = { 's' ?1kb:0xf };   ExpectedValue = 1kb }
+                @{ Script = { $null ?1kb:0xf }; ExpectedValue = 15 }
+                @{ Script = { '' ?1kb:0xf };    ExpectedValue = 15 }
+
+                ## Condition: other primary expressions
+                @{ Script = { 1,2,3,4 ? 'Core' : 'Desktop' };          ExpectedValue = 'Core' }
+                @{ Script = { @(1,2,3,4) ? 'Core' : 'Desktop' };       ExpectedValue = 'Core' }
+                @{ Script = { @{name = 'name'} ? 'Core' : 'Desktop' };  ExpectedValue = 'Core' }
+                @{ Script = { @{name = 'name'}.name ? 'Core' : 'Desktop' }; ExpectedValue = 'Core' }
+                @{ Script = { @{name = 'name'}.Contains('name') ? 'Core' : 'Desktop' }; ExpectedValue = 'Core' }
+                @{ Script = { (Test-Path Env:\NonExist) ? 'true' : 'false' };     ExpectedValue = 'false' }
+                @{ Script = { (Test-Path Env:\PSModulePath) ? 'true' : 'false' }; ExpectedValue = 'true' }
```

Comment:
If my PR for PSModulePath will accepted we can haven't the environment variable at all.

Reply:
Removing this environment variable sounds like a big breaking change to me ... Anyways, that's a separate issue.